astroquery/hips2fits/tests/test_hips2fits_remote.py

```diff
@@ -86,8 +83,8 @@ def test_query_jpg_no_wcs(self):
         assert isinstance(result, np.ndarray) and result.shape[2] == 3
 
     def test_bad_strech(self):
-        with pytest.raises(AttributeError):
-            result = hips2fits.query_with_wcs(
+        with pytest.raises(AstropyUserWarning, match=r"stretch: must either"):
```

Comment:
This should not be changed, I still see an attribute error raised by this test.

```
E           AttributeError: Unknown value azs for stretch parameter
```

Reply:
I'll revert the change.  I see I was testing with `-W error` to try to figure out why I was seeing some warnings/errors in some tests.

So this is where testing with `-W error` or similar turning warnings to errors in `setup.cfg` makes this difficult.  If underlying library code returns an error, but the library wrapper handles it, or if some internal code returns a warning, this doesn't raise an exception.  If warnings are turned into exceptions in a blanket way, the test fails with that internal warning-turned-exception.  If some warnings are allowed, it returns a different exception.

So for this test:

```
$ pytest -rsxf astroquery/hips2fits --remote-data=any -k bad_strech
========================= test session starts =========================
platform darwin -- Python 3.10.4, pytest-7.1.3, pluggy-1.0.0

Running tests with astroquery version 0.4.7.dev8130_testrun_testrun.
...
rootdir: /Users/jdavies/dev/astroquery, configfile: setup.cfg
plugins: hypothesis-6.54.6, remotedata-0.3.3, astropy-header-0.2.2, mock-3.8.2, dependency-0.5.1, astropy-0.10.0, filter-subpackage-0.1.1, openfiles-0.5.0, doctestplus-0.12.0, cov-3.0.0, arraydiff-0.5.0
collected 10 items / 9 deselected / 1 selected                        

astroquery/hips2fits/tests/test_hips2fits_remote.py .           [100%]

=================== 1 passed, 9 deselected in 0.62s ===================
```

So `AttributeError` is raised and the test passes.  As intended in the test.  But

```
$ pytest -rsxf astroquery/hips2fits --remote-data=any -k bad_strech -W error
========================= test session starts =========================
platform darwin -- Python 3.10.4, pytest-7.1.3, pluggy-1.0.0
...
rootdir: /Users/jdavies/dev/astroquery, configfile: setup.cfg
plugins: hypothesis-6.54.6, remotedata-0.3.3, astropy-header-0.2.2, mock-3.8.2, dependency-0.5.1, astropy-0.10.0, filter-subpackage-0.1.1, openfiles-0.5.0, doctestplus-0.12.0, cov-3.0.0, arraydiff-0.5.0
collected 10 items / 9 deselected / 1 selected                        

astroquery/hips2fits/tests/test_hips2fits_remote.py F           [100%]

============================== FAILURES ===============================
_________________ TestHips2fitsRemote.test_bad_strech _________________

self = <astroquery.hips2fits.tests.test_hips2fits_remote.TestHips2fitsRemote object at 0x7fd72229f1f0>

    def test_bad_strech(self):
        with pytest.raises(AttributeError, match=r"Unknown value azs for stretch parameter"):
>           hips2fits.query_with_wcs(

...

>           warnings.warn(msg, AstropyUserWarning)
E           astropy.utils.exceptions.AstropyUserWarning: stretch: must either 'power', 'linear', 'sqrt', 'log' or 'asinh'.
E           azs is ignored.

astroquery/hips2fits/core.py:377: AstropyUserWarning
======================= short test summary info =======================
FAILED astroquery/hips2fits/tests/test_hips2fits_remote.py::TestHips2fitsRemote::test_bad_strech
=================== 1 failed, 9 deselected in 0.53s ===================
```

And this is because `AstropyUserWarning` is allowed in `setup.cfg`.  The difficulty of turning warnings into errors.  Of course both are related to stretch, so it's not clear to me if one is better to handle than the other.  Maybe both?  One could do `with pytest.raises((AttributeError, AstropyUserWarning)):` and it would allow both.

Which is preferred?

Reply:
`AstropyUserWarning` should not be allowed, and no blanket rules should be included for overriding errors/warnings in setup.cfg. I already have a  PR that fixes this but it run into a few complicated cases (and possibly some conflicts with PRs merged in since then).

> The difficulty of turning warnings into errors. Of course, both are related to stretch, so it's not clear to me if one is better to handle than the other. Maybe both? One could do with pytest.raises((AttributeError, AstropyUserWarning)): and it would allow both.

I think it's wrong to allow both. I think the tests should capture expected errors/warnings that the user would see, but don't work around the ones generated by the test/docs build procedures.

Reply:
Ok, thanks.  I've reverted my changes here and in the test below.  👍

Reply:
so if any of the errors remain, we can "todo" list them, and do a temporary xfail for them, (or temporary setup.cfg if nothing else works). How that that sounds?